src/lib/Cropper.svelte

```diff
@@ -229,34 +230,55 @@
       restrictPosition
     )
 
-    dispatch('cropcomplete', {
+    oncropcomplete?.({
       percent: croppedAreaPercentages,
       pixels: croppedAreaPixels,
     })
   }
 
   // ------ Reactive statement ------
   //when aspect changes, we reset the cropperSize
-  $: if (imgEl) {
-    cropperSize = cropSize ? cropSize : helpers.getCropSize(imgEl.width, imgEl.height, aspect)
-  }
+  $effect(() => {
+    if (imgEl) {
+      cropperSize = cropSize ? cropSize : helpers.getCropSize(imgEl.width, imgEl.height, aspect)
+    }
+  })
 
   // when zoom changes, we recompute the cropped area
-  $: zoom && emitCropData()
+  let lastZoom = $state(zoom);
+  $effect(() => {
+    if (zoom !== lastZoom) {
+      lastZoom = zoom;
+      emitCropData()
+    }
+  })
 
-  $: if(aspect) {
-    computeSizes()
-    emitCropData()
-  } 
+  let lastAspect = $state(aspect);
```

Comment:
Same here, this state should not be needed.

Reply:
The `lastAspect` is required here, when checking for just the `aspect` variable, it enters an infinite loop.